onnx/optimizer/optimize.h

```diff
@@ -32,9 +32,10 @@ struct Optimizer {
     passes[sp->name] = std::move(sp);
   }
 
-  std::unique_ptr<onnx::ModelProto> optimize(std::unique_ptr<onnx::ModelProto> mp_in, std::vector<std::string>& names) {
-
-    std::shared_ptr<onnx::Graph> g(onnx::ImportModelProto(*mp_in));
+  onnx::ModelProto optimize(
+      const onnx::ModelProto& mp_in,
```

Comment:
At least, we have one copy here, right? (Although this ModelProto is a reference, the instance itself will live until the end of this function.)

Reply:
You have a point here. @bddppq, the point is `mp_in` is useless right after 
```
 std::shared_ptr<onnx::Graph> g(onnx::ImportModelProto(*mp_in));
```
and can thus be released to reduce memory. Therefore, it should not be `const`.

Reply:
If we can change the optimizer to do in-place, then we should just change the const ref to ref in this PR, instead of passing in a copy through unique_ptr, delete it and then recreate another mp_out to return.